src/main/java/com/opentable/kafka/util/OffsetMetrics.java

```diff
@@ -0,0 +1,242 @@
+                metricPrefix,
+                metricRegistry,
+                groupId,
+                brokerList
+        );
+    }
+
+    OffsetMetrics(
+            final String metricPrefix,
+            final MetricRegistry metricRegistry,
+            final String groupId,
+            final String brokerList,
+            final Collection<String> topics,
+            final Supplier<Reservoir> reservoirSupplier,
+            final Duration pollPeriod) {
+        Preconditions.checkArgument(metricPrefix != null, "null metric prefix");
+        Preconditions.checkArgument(!topics.isEmpty(), "no topics");
+        this.metricPrefix = metricPrefix;
+        this.metricRegistry = metricRegistry;
+        this.groupId = groupId;
+        this.topics = topics;
+        this.pollPeriod = pollPeriod;
+        monitor = new OffsetMonitor(groupId, brokerList);
+
+        final Collection<String> badTopics = new ArrayList<>();
+        final ImmutableMap.Builder<String, Metric> builder = ImmutableMap.builder();
+        topics.forEach(topic -> {
+            final Map<Integer, Long> sizes = monitor.getTopicSizes(topic);
+            if (sizes.isEmpty()) {
+                badTopics.add(topic);
+                return;
+            }
+            for (final int part : sizes.keySet()) {
+                builder.put(partitionName(topic, part, "size"), new AtomicLongGauge());
+                builder.put(partitionName(topic, part, "offset"), new AtomicLongGauge());
+                builder.put(partitionName(topic, part, "lag"), new AtomicLongGauge());
+            }
+            builder.put(totalName(topic, "size"), new AtomicLongGauge());
+            builder.put(totalName(topic, "offset"), new AtomicLongGauge());
+            builder.put(totalName(topic, "lag"), new AtomicLongGauge());
+
+            builder.put(totalName(topic, "lag-distribution"), new Histogram(reservoirSupplier.get()));
+        });
+
+        if (!badTopics.isEmpty()) {
+            throw new IllegalArgumentException(
+                    String.format("bad topic%s: '%s'", badTopics.size() != 1 ? "s" : "", badTopics));
+        }
+
+        metricMap = builder.build();
+
+        exec = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setNameFormat(String.format("%s-offset-metrics-%%d", groupId))
+                        .build()
+        );
+    }
+
+    private String partitionName(final String topic, final int partition, final String name) {
+        return String.format("%s.%s.partition.%d.%s", metricPrefix, topic, partition, name);
+    }
+
+    private String totalName(final String topic, final String name) {
+        return String.format("%s.%s.total.%s", metricPrefix, topic, name);
+    }
+
+    @VisibleForTesting
+    Map<String, Metric> getMetrics() {
+        return metricMap;
+    }
+
+    @PostConstruct
+    public void start() {
+        metricRegistry.registerAll(() -> metricMap);
```

Comment:
totally tiny nit, but you could consider `this::getMetrics` instead of a lambda (preferring a named entity to an anonymous one)

Reply:
Ooh, nice.  Implemented in f40bd81afe8bd1cdc90072e8f55355825729a746.